test/unit/org/openstreetmap/josm/data/imagery/WMTSTileSourceTest.java

```diff
@@ -230,17 +241,39 @@ public void testTicket12168() throws IOException {
     }
 
     @Test
-    @Ignore("disabled as this needs user action") // XXX
     public void testTwoTileSetsForOneProjection() throws Exception {
         Main.setProjection(Projections.getProjectionByCode("EPSG:3857"));
-        WMTSTileSource testSource = new WMTSTileSource(testImageryOntario);
+        ImageryInfo ontario = getImagery(TestUtils.getTestDataRoot() + "wmts/WMTSCapabilities-Ontario.xml");
+        ontario.setDefaultLayers(Arrays.asList(new DefaultLayer[] {
+                new DefaultLayer(ImageryType.WMTS, "Basemap_Imagery_2014", null, "default028mm")
+        }));
+        WMTSTileSource testSource = new WMTSTileSource(ontario);
+        testSource.initProjection(Main.getProjection());
+        assertEquals(
+                "http://maps.ottawa.ca/arcgis/rest/services/Basemap_Imagery_2014/MapServer/WMTS/tile/1.0.0/Basemap_Imagery_2014/default/default028mm/4/2932/2371.jpg",
+                testSource.getTileUrl(4, 2371, 2932));
+        verifyTile(new LatLon(45.4601306, -75.7617187), testSource, 2372, 2932, 4);
+        verifyTile(new LatLon(45.4602510, -75.7617187), testSource, 607232, 750591, 12);
+    }
+
+    @Test
+    public void testTwoTileSetsForOneProjectionSecondLayer() throws Exception {
+        Main.setProjection(Projections.getProjectionByCode("EPSG:3857"));
+        ImageryInfo ontario = getImagery(TestUtils.getTestDataRoot() + "wmts/WMTSCapabilities-Ontario.xml");
+        ontario.setDefaultLayers(Arrays.asList(new DefaultLayer[] {
+                new DefaultLayer(ImageryType.WMTS, "Basemap_Imagery_2014", null, "GoogleMapsCompatible")
+        }));
+        WMTSTileSource testSource = new WMTSTileSource(ontario);
         testSource.initProjection(Main.getProjection());
-        verifyTile(new LatLon(45.4105023, -75.7153702), testSource, 303751, 375502, 12);
-        verifyTile(new LatLon(45.4601306, -75.7617187), testSource, 1186, 1466, 4);
+        assertEquals(
+                "http://maps.ottawa.ca/arcgis/rest/services/Basemap_Imagery_2014/MapServer/WMTS/tile/1.0.0/Basemap_Imagery_2014/default/GoogleMapsCompatible/4/2932/2371.jpg",
+                testSource.getTileUrl(4, 2371, 2932));
+        verifyMercatorTile(testSource, 74, 91, 8);
+        verifyMercatorTile(testSource, 37952, 46912, 17);
     }
 
     @Test
-    @Ignore("disabled as this needs user action") // XXX
+    //@Ignore("disabled as this needs user action") // XXX
```

Comment:
if the test does not need user action any more, the line should be deleted, not commented